app/models/product_user_observer.rb

```diff
@@ -0,0 +1,15 @@
+class ProductUserObserver < ActiveRecord::Observer
```

Comment:
I've become less of a fan of observers like this over time. From what I've seen they tend to be almost more trouble than callbacks, and even harder to realize that they're there.

What would you think of a `ProductUserCreator` class that handles this, and use that inside the ProductUserController#create action (which confusingly seems to be the `new` method)?